.github/workflows/release.yml

```diff
@@ -68,16 +72,31 @@ jobs:
         run: Remove-Item -LiteralPath "C:\msys64\" -Force -Recurse
         if: runner.os == 'Windows'
 
-      - name: AArch64 cross-compile packages
+      - name: Linux AArch64 cross-compile packages
         run: sudo apt-get install -y --no-install-recommends g++-aarch64-linux-gnu gcc-aarch64-linux-gnu libc6-dev-arm64-cross
         if: matrix.build.target == 'aarch64-unknown-linux-gnu'
 
-      - name: Build the executable
+      - name: Build the executable (other than aarch64 linux)
         uses: actions-rs/cargo@ae10961054e4aa8b4aa7dffede299aaf087aa33b # @v1.0.1
+        if: matrix.build.target != 'aarch64-unknown-linux-gnu'
         with:
           command: build
           args: --locked -Z build-std --target ${{ matrix.build.target }} --profile production --bin subspace-cli
 
+      # TODO: get rid of this when we have bigger RAM for aarch64 linux
+      - name: Build the executable for aarch64 linux separately
+        uses: actions-rs/cargo@ae10961054e4aa8b4aa7dffede299aaf087aa33b # @v1.0.1
+        if: matrix.build.target == 'aarch64-unknown-linux-gnu'
+        with:
+          command: build
+          args: --locked -Z build-std --target ${{ matrix.build.target }} --profile aarch64linux --bin subspace-cli
+
+      # TODO: get rid of this when we have bigger RAM for aarch64 linux
+      - name: Rename the aarch64 linux target folder to `production` for compatibility
+        if: matrix.build.target == 'aarch64-unknown-linux-gnu'
+        run: |
+          mv target/${{ matrix.build.target }}/aarch64linux ${{ env.PRODUCTION_TARGET }}
+
```

Comment:
`TARGET_DIR` is: `target/${{ matrix.build.target }}/production`

> and you essentially moving `target/aarch64-unknown-linux-gnu/aarch64linux` to target/production/
No, I'm renaming:
`target/aarch64-unknown-linux-gnu/aarch64linux` - `target/aarch64-unknown-linux-gnu/production`

Since, in the later workflow, we are using `production` folder

Reply:
Okay then. I would just set `env.PRODUCTION_TARGET` to `target/aarch64-unknown-linux-gnu/aarch64linux` for aarch64 and remove this step.

Reply:
you mean a step like

```
- name: rename production target variable for aarch64 linux
   if: matrix.build.target == 'aarch64-unknown-linux-gnu'
   run: |
     {{ env.PRODUCTION_TARGET }} = target/aarch64-unknown-linux-gnu/aarch64linux
```

i do not think this is a better solution, in fact, this seems odder to me. And either way, 
potato vs potato :)

Reply:
No, I mean set it in matrix for each of the targets

Reply:
Oh ok! Yeah that should be more pretty, lemme do that real quick 👍